lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -913,24 +1242,32 @@ def from_ansible_params(ap, old_service, image_digest, can_update_networks):
 
         s.env = get_docker_environment(ap['env'], ap['env_files'])
 
-        s.restart_policy_delay = get_nanoseconds_from_raw_option(
-            'restart_policy_delay',
-            ap['restart_policy_delay']
-        )
-        s.restart_policy_window = get_nanoseconds_from_raw_option(
-            'restart_policy_window',
-            ap['restart_policy_window']
-        )
+        update_config = cls.get_update_config_from_ansible_params(ap)
```

Comment:
Alternatively, you could also chang ethe `get_xxx_from_ansible_params` to also get `s`, and change them to directly update `s`. Then you wouldn't need this loop after each call. But both ways are fine for me.

Reply:
Then let's keep it as it's today. In my opinion it's nicer to keep most functions as "pure" as possible and mutate state in one place.

Reply:
Fine for me. I in general also prefer pure functions, but I prefer non-pure functions over `getattr`/`setattr` magic ;)